src/backend/src/lib.rs

```diff
@@ -154,36 +155,60 @@ pub struct Config {
     pub ecdsa_key_name: String,
     // A list of allowed callers to restrict access to endpoints that do not particularly check or use the caller()
     pub allowed_callers: Vec<Principal>,
+    pub supported_credentials: Option<Vec<SupportedCredential>>,
+    /// Root of trust for checking canister signatures.
+    pub ic_root_key_raw: Option<Vec<u8>>,
+}
+
+fn set_config(arg: InitArg) {
+    let InitArg {
+        ecdsa_key_name,
+        allowed_callers,
+        supported_credentials,
+        ic_root_key_der,
+    } = arg;
+    mutate_state(|state| {
+        let ic_root_key_raw =
```

Comment:
With the help of ChatGPT, if interesting, the two lines version of this block:

```
let ic_root_key_raw = extract_raw_root_pk_from_der(
    &ic_root_key_der.unwrap_or_else(|| IC_ROOT_PK_DER.to_vec())
).map(Some).unwrap_or_else(|_| panic!("There was an error parsing the root key"));
```

Reply:
🎉